integrations/integrations.go

```diff
@@ -0,0 +1,231 @@
+package integrations
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net"
+	"os"
+	"os/signal"
+	"strconv"
+	"strings"
+	"syscall"
+
+	cometdb "github.com/cometbft/cometbft-db"
+	cometconfig "github.com/cometbft/cometbft/config"
+	dbm "github.com/cosmos/cosmos-db"
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/server"
+	serverconfig "github.com/cosmos/cosmos-sdk/server/config"
+	servertypes "github.com/cosmos/cosmos-sdk/server/types"
+	genutiltypes "github.com/cosmos/cosmos-sdk/x/genutil/types"
+	"github.com/polymerdao/monomer"
+	"github.com/polymerdao/monomer/environment"
+	"github.com/polymerdao/monomer/genesis"
+	"github.com/polymerdao/monomer/node"
+	"github.com/spf13/viper"
+)
+
+const (
+	monomerGenesisPathFlag = "monomer-genesis-path"
+	monomerEngineWSFlag    = "monomer-engine-ws"
+)
+
+// StartCommandHandler is a custom callback that overrides the default `start` function in the Cosmos
+// SDK. It starts a Monomer node in-process instead of a CometBFT node.
+func StartCommandHandler(
+	svrCtx *server.Context,
+	clientCtx client.Context, //nolint:gocritic // hugeParam
+	appCreator servertypes.AppCreator,
+	inProcessConsensus bool,
+	opts server.StartCmdOptions,
+) error {
+	// We assume `inProcessConsensus` is true for now, so let's return an error if it's not.
+	if !inProcessConsensus {
+		return fmt.Errorf("in-process consensus must be enabled")
+	}
+
+	env := environment.New()
+	defer func() {
+		if err := env.Close(); err != nil {
+			svrCtx.Logger.Error("Failed to close environment", "err", err)
```

Comment:
My bad, I forgot we can handle this error and pass it to the caller instead of punting and printing a log. I have some helper code for this in a separate PR, so we can ignore it for now.